srcs/http/HTTPHandler.cpp

```diff
@@ -12,28 +12,37 @@ void http_handle(smt::shared_ptr<ServerSocket> sock, int client_fd) {
     HTTPParser                   parser;
     smt::shared_ptr<HTTPRequest> request = parser.getNextRequest(req_str);
     bool                         has_next = true;
-    while (has_next) {
+    while (has_next && request->isValid()) {
 
-        FLOG("Handling request...");
-        // // handle request here
-        // smt::shared_ptr<HTTPResponse> response =
-        //     process_request(request, sock->m_config);
-        //
-        // sock->send(client_fd, response->to_str()); // sending response to
-        // client
+        LOG_D("Handling request...");
+        // Find the appropriate ServerBlock
+        int serverBlockIdx = sock->bestServerBlockForRequest(request);
+        serverBlockIdx = (serverBlockIdx == -1 ? 0 : serverBlockIdx);
+        // handle request here
+        smt::shared_ptr<HTTPResponse> response = process_request(request, sock->m_blocks[serverBlockIdx], client_fd); // TODO: This obviously needs to be fixed, just using the first block here so that rest of code can run
```

Comment:
In case no block is found, we will use the default block here. Which is the first server block on the config file. So I'm not sure how wrong this is

Reply:
I don't think I'm 100% correct here. I know its the default block will be the first of something, not sure if it is the first of the entire file